profiler/src/ProfilerEngine/Datadog.Linux.ApiWrapper/functions_to_wrap.c

```diff
@@ -59,13 +60,210 @@ unsigned long long dd_inside_wrapped_functions()
+{
+    if (access("/etc/alpine-release", F_OK) == 0)
+    {
+        return 1;
+    }
+
+    return 0;
+}
+
+char* appendToPath(const char* folder, const char* suffix)
+{
+    int length = strlen(folder) + strlen(suffix);
+
+    char* result = malloc(length + 1);
+
+    if (result == NULL)
+    {
+        return NULL;
+    }
+
+    strcpy(result, folder);
+    strcpy(result + strlen(folder), suffix);
+
+    return result;
+}
+
+char* getFolder(const char* path)
+{
+    char* lastSlash = strrchr(path, '/');
+    if (lastSlash == NULL)
+    {
+        return "";
+    }
+
+    // Copying the last subfolder to a new string
+    int length = lastSlash - path;
+    char* folder = malloc(length + 1);
+
+    if (folder == NULL)
+    {
+        return NULL; // Memory allocation failed
+    }
+
+    strncpy(folder, path, length);
+    folder[length] = '\0';
+
+    return folder;
+}
+
+char* getSubfolder(const char* path)
+{
+    char* lastSlash = strrchr(path, '/');
+    if (lastSlash == NULL) {
+        return NULL; // No slash found, path does not contain subfolders
+    }
+
+    // Finding the last but one slash
+    char* temp = lastSlash - 1;
+    while (temp >= path && *temp != '/') {
+        temp--;
+    }
+
+    if (temp < path)
+    {
+        // No leading slash, that's weird
+        return NULL;
+    }
+
+    // Increment to move past the slash
+    temp++;
+
+    // Copying the last subfolder to a new string
+    int folderLength = lastSlash - temp;
+    char* subfolder = malloc(folderLength + 1);
+
+    if (subfolder == NULL)
+    {
+        return NULL; // Memory allocation failed
+    }
+
+    strncpy(subfolder, temp, folderLength);
+    subfolder[folderLength] = '\0';
+
+    return subfolder;
+}
+
 __attribute__((constructor))
-void initLibrary(void) {
+void initLibrary(void)
+{
+    const char* crashHandlerEnabled = getenv("DD_TRACE_CRASH_HANDLER_ENABLED");
+
+    if (crashHandlerEnabled != NULL)
+    {
+        if (strcasecmp(crashHandlerEnabled, "no") == 0
+         || strcasecmp(crashHandlerEnabled, "false") == 0
+         || strcasecmp(crashHandlerEnabled, "0") == 0)
+        {
+            // Early nope
+            return;
+        }
+    }
+
     // If crashtracking is enabled, check the value of DOTNET_DbgEnableMiniDump
     // If set, set DD_TRACE_CRASH_HANDLER_PASSTHROUGH to indicate dd-dotnet that it should call createdump
     // If not set, set it to 1 so that .NET calls createdump in case of crash
     // (and we will redirect the call to dd-dotnet)
-    char* crashHandler = getenv("DD_TRACE_CRASH_HANDLER");
+    const char* crashHandlerEnv = getenv("DD_TRACE_CRASH_HANDLER");
+
+    if (crashHandlerEnv == NULL || crashHandlerEnv[0] == '\0')
+    {
+        // The path to the crash handler is not set, try to deduce it
+        const char* libraryPath = getLibraryPath();
+
+        if (libraryPath != NULL)
+        {            
+            // If the library is in linux-x64 or linux-musl-x64, we use that folder
```

Comment:
We could also be in the root folder, in which case we should also call `isAlpine()` and use the correct folder

~~Also, we should handle arm64 too I think?~~ I see, we do, it's just the comments that don't mention it

Reply:
Done in https://github.com/DataDog/dd-trace-dotnet/pull/5582/commits/b2e7da337feb094d5b715222f6343b75d486fa58